PCF-ADR-0008-ray-cluster-authentication.md

```diff
@@ -0,0 +1,94 @@
+# Project CodeFlare - RayCluster Authentication
+|                |                                                                                  |
+| -------------- | -------------------------------------------------------------------------------- |
+| Date           | 12/19/2023                                                                       |
+| Scope          |                                                                                  |
+| Status         | provisional                                                                      |
+| Authors        | [Kevin Postlethwait](@KPostOffice)                                               |
+| Supersedes     | N/A                                                                              |
+| Superseded by: | N/A                                                                              |
+| Issues         |                                                                                  |
+| Other docs:    | none                                                                             |
+
+## What
+
+Enabling OpenShift OAuth as an identity provider and authorization mechanism for KubeRay RayClusters.
+
+## Why
+
+Currently there is no authentication by default enabled in front of the Ray Dashboard and we would like to offer a
+unified login experience for the whole of ODH by way of OpenShift OAuth
+
+## Goals
+
+* Protect the dashboard from unauthenticated access by way of an OAuth flow
+* Protect the API endpoint from unauthenticated access by way of bearer tokens
+* Enable HTTPS by default for accessing the RayCluster API and Dashboard
+* Support authenticated interactive session
+
+## Non-Goals
+
+* Creating a fully customizable auth(z) solution
+
+## How
+
+In order to solve this we will add a RayCluster reconciler which creates the necessary resources for enabling
+enabling authentication and authorization in front of the API and dashboard.
+
+![Auth Diagram](images/PCF-ADR-008-ray-cluster-auth.jpg)
+
+This will change in Q1 or early Q2 of 2024 with the introduction of Authorino. The overall design stays the same, but
+the resources created by the CodeFlare Operator's RayCluster reconciler will change and there will likely no longer be
```

Comment:
Did you mean KubeRay operator? 
```suggestion
the resources created by the KubeRay Operator's RayCluster reconciler will change and there will likely no longer be
```

Reply:
Never mind. I missed some context in the conversation. You will add a CodeFlare Operator's RayCluster reconciler in the CFO.